jenkins-design-language/src/js/components/status/SvgSpinner.jsx

```diff
@@ -6,6 +6,33 @@ import { describeArcAsPath } from '../SVG';
 export const strokeWidth = 3.5; // px. Maybe we can fetch this from CSS at runtime in the future
 
 export default class SvgSpinner extends Component {
+    componentWillMount() {
+        this.infiniteRotationRunning = false;
+        this.setState({
+            infiniteRotatePercentage: 0
+        });
+    }
+
+    infiniteLoadingTimer = () => {
```

Comment:
I suggest recording the time elapsed with `infiniteLoadingTimer = (currentTime) => {`....